sw/device/tests/dif/dif_aon_timer_unittest.cc

```diff
@@ -0,0 +1,494 @@
+
+TEST_F(WatchdogGetCountTest, NullArgs) {
+  EXPECT_EQ(dif_aon_timer_watchdog_get_count(nullptr, nullptr),
+            kDifAonTimerWatchdogBadArg);
+  EXPECT_EQ(dif_aon_timer_watchdog_get_count(&aon_, nullptr),
+            kDifAonTimerWatchdogBadArg);
+  uint32_t count;
+  EXPECT_EQ(dif_aon_timer_watchdog_get_count(nullptr, &count),
+            kDifAonTimerWatchdogBadArg);
+}
+
+TEST_F(WatchdogGetCountTest, Success) {
+  EXPECT_READ32(AON_TIMER_WDOG_COUNT_REG_OFFSET, 0xA5A5A5A5);
+
+  uint32_t count;
+  EXPECT_EQ(dif_aon_timer_watchdog_get_count(&aon_, &count),
+            kDifAonTimerWatchdogOk);
+  EXPECT_EQ(count, 0xA5A5A5A5);
+}
+
+class WatchdogPetTest : public AonTimerTest {};
+
+TEST_F(WatchdogPetTest, NullArgs) {
+  EXPECT_EQ(dif_aon_timer_watchdog_pet(nullptr), kDifAonTimerBadArg);
+}
+
+TEST_F(WatchdogPetTest, Success) {
+  EXPECT_WRITE32(AON_TIMER_WDOG_COUNT_REG_OFFSET, 0);
+
+  EXPECT_EQ(dif_aon_timer_watchdog_pet(&aon_), kDifAonTimerOk);
+}
+
+class WatchdogLockTest : public AonTimerTest {};
+
+TEST_F(WatchdogLockTest, NullArgs) {
+  EXPECT_EQ(dif_aon_timer_watchdog_lock(nullptr), kDifAonTimerBadArg);
+}
+
+TEST_F(WatchdogLockTest, Success) {
+  // Write one to clear.
+  EXPECT_WRITE32(AON_TIMER_WDOG_REGWEN_REG_OFFSET, 1);
+
+  EXPECT_EQ(dif_aon_timer_watchdog_lock(&aon_), kDifAonTimerOk);
+}
+
+class WatchdogIsLockedTest : public AonTimerTest {};
+
+TEST_F(WatchdogIsLockedTest, NullArgs) {
+  EXPECT_EQ(dif_aon_timer_watchdog_is_locked(nullptr, nullptr),
+            kDifAonTimerBadArg);
+  EXPECT_EQ(dif_aon_timer_watchdog_is_locked(&aon_, nullptr),
+            kDifAonTimerBadArg);
+  bool is_locked;
+  EXPECT_EQ(dif_aon_timer_watchdog_is_locked(nullptr, &is_locked),
+            kDifAonTimerBadArg);
+}
+
+TEST_F(WatchdogIsLockedTest, Success) {
+  EXPECT_READ32(AON_TIMER_WDOG_REGWEN_REG_OFFSET, 1);
+
+  bool is_locked;
+  EXPECT_EQ(dif_aon_timer_watchdog_is_locked(&aon_, &is_locked),
+            kDifAonTimerOk);
+  EXPECT_EQ(is_locked, false);
+}
+
+TEST_F(WatchdogIsLockedTest, SuccessLocked) {
+  EXPECT_READ32(AON_TIMER_WDOG_REGWEN_REG_OFFSET, 0);
+
+  bool is_locked;
+  EXPECT_EQ(dif_aon_timer_watchdog_is_locked(&aon_, &is_locked),
+            kDifAonTimerOk);
+  EXPECT_EQ(is_locked, true);
+}
+
+class IrqIsPendingTest : public AonTimerTest {};
+
+TEST_F(IrqIsPendingTest, NullArgs) {
+  EXPECT_EQ(dif_aon_timer_irq_is_pending(
+                nullptr, kDifAonTimerIrqWakeupThreshold, nullptr),
+            kDifAonTimerBadArg);
+  EXPECT_EQ(dif_aon_timer_irq_is_pending(&aon_, kDifAonTimerIrqWakeupThreshold,
+                                         nullptr),
+            kDifAonTimerBadArg);
+  bool is_pending;
+  EXPECT_EQ(dif_aon_timer_irq_is_pending(
+                nullptr, kDifAonTimerIrqWakeupThreshold, &is_pending),
+            kDifAonTimerBadArg);
+}
+
+TEST_F(IrqIsPendingTest, BadInterrupt) {
+  bool is_pending;
+  int invalid = kDifAonTimerIrqWatchdogBarkThreshold;
+  ++invalid;
```

Comment:
Why not `kVariant + 1`...?

Reply:
C++ doesn't seem to let you do it that way

Reply:
`int valid = static_cast<int>(...) + 1;`.

Just a matter of knowing the language. Let's try to avoid doing things that might confuse a reader like this that rely on implicit conversion.

Reply:
The way I done it is just a different way of doing it, but I agree your suggestion is better.

Reply:
Done.